### ISSUE TITLE
Lots of bugs in the program

### ISSUE THREAD

**Issue by KennthShang** [crazyinter/Gene2Vec#2]
Hi, thanks for providing Virtifier.

However, it seems that a lot of bugs exist in this GitHub folder, including training, testing, and preprocessing files. 

Also, it seems the program is not an end-to-end pipeline, which only allows for evaluating the performance with labeled data rather than analyzing the contigs from metagenomic data.

What's more, I suppose that a more comprehensive README file should be provided to introduce how to use your program such as some testing samples for users to get started. Although there are some kmer features in the data folder, the labels are missing.

BTW, it seems that the attention function has never been called in the program. According to the paper, I think maybe attention is the most important part of Virtifier? Without the attention, it is just a simple RNN model.

Best,
Jiayu

**Comment by crazyinter** [comment]
Hi, thanks for your interests to Virtifier.

As what you have said, we are sorry to say that the program is not an end-to-end pipeline right now. It is only allowed for evaluating the performance with labeled data. We are working on making it available as an end-to-end pipeline so that everyone can analyze the contigs from metagenomic data. 

In the data folder, there are not kmer features. They are viral and host contigs being prreprocessed into sequences of codons. The numbers are the order of every codon in the codon dictionary so that it is easy to be embedded by the embedding matrix.

The file "train_500bp.csv" contains 4500 viral sequences and 4500 host sequences of length 500bp as described in README file, so we didn't provide the labels. The first 4500 sequences are viruses, and the last 4500 sequences are host. It is easy to generate when you try to use Virtifier.

The attention mechanism and the codon embedding are both the most important part of Virtifier. I am sorry that the attention function was not called in the program because of my mistakes. I have added it into the 'dynamicRNN()'.

**Comment by crazyinter** [comment]
Hi, thanks for your interests to Virtifier.

As what you have said, we are sorry to say that the program is not an end-to-end pipeline right now. It is only allowed for evaluating the performance with labeled data. We are working on making it available as an end-to-end pipeline so that everyone can analyze the contigs from metagenomic data. 

In the data folder, there are not kmer features. They are viral and host contigs being prreprocessed into sequences of codons. The numbers are the order of every codon in the codon dictionary so that it is easy to be embedded by the embedding matrix.

The file "train_500bp.csv" contains 4500 viral sequences and 4500 host sequences of length 500bp as described in README file, so we didn't provide the labels. The first 4500 sequences are viruses, and the last 4500 sequences are host. It is easy to generate when you try to use Virtifier.

The attention mechanism and the codon embedding are both the most important part of Virtifier. I am sorry that the attention function was not called in the program because of my mistakes. I have added it into the 'dynamicRNN()'. 

If you have any other questions about Virtifier, it is my pleasure to be informed.

Best regards,
Yan Miao

**Comment by KennthShang** [comment]
Thanks for your updates.

However, it seems the program still can not be run, and lots of bugs occur in this new version.

BTW, I am using the mentioned env "tensorflow=1.3"

Best,
Jiayu

**Comment by crazyinter** [comment]
我的环境是python3.6.5+tf1.2.0，不知道你运行的时候具体在哪步出现了错误呢？这些代码都是我从jupyternotebook上粘过来的，或者你可以尝试一下jupyter notebook，分块运行一下，更利于发现错误及调试。之后联系我可以通过邮件[miaoyan17@mails.jlu.edu.cn](mailto:miaoyan17@mails.jlu.edu.cn)，github的消息不会经常阅览，而且我最近事情比较忙，可能不能及时回复，还请见谅。

**Comment by KennthShang** [comment]
你好，

在README文件中，你推荐的TF版本号是1.3.0

以及你有尝试过先运行一下你的Train.py和Test.py文件吗？

Best,
Jiayu

**Comment by crazyinter** [comment]
您好，
tf1.3.0和1.2.0应该都可以使用，因为我这里有不同的环境的电脑，所以两个版本都用过。上传到github的代码Train.py和Test.py文件是我从jupyternotebook上逐段贴过来的，没有重新运行过，这几天我会找时间运行一下Train.py和Test.py文件，维护一下其中的bug
